### ISSUE TITLE
Add support for concatMapEager

### ISSUE THREAD

**Issue by rochdev** [Reactive-Extensions/RxJS#1499]
When calling `concatMap`, the handler is called in series so it has to return a value before entering the next iteration.

In some cases it would be more efficient to be able to run the handler in parallel and then drain the results in order, especially if the handler is waiting for IO.

See http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#concatMapEager(io.reactivex.functions.Function)

**Comment by rochdev** [comment]
Closing in favor of https://github.com/ReactiveX/rxjs/issues/2733
